### PR TITLE
SWC-5555: explicitly add the modal to the DOM (in the files tab view)…

### PR DESCRIPTION
**Pull request by jay-hodgson**
…, and clean up if detached

### CODE REVIEW

File: src/main/java/org/sagebionetworks/web/client/widget/entity/tabs/FilesTabViewImpl.java [hunk 1]
```diff
@@ -85,33 +87,35 @@ public FilesTabViewImpl(UserBadge createdByBadge, UserBadge modifiedByBadge) {
...
+		return event -> {
+			modalDialogContainer.clear();
+			final Modal window = new Modal();
+			window.addStyleName("modal-fullscreen");
+			final ModalBody body = new ModalBody();
+			final Div oldParent = (Div) w.getParent();
+			w.removeFromParent();
+			body.add(new ScrollPanel(w));
+			w.setHeight(new Double(com.google.gwt.user.client.Window.getClientHeight()).intValue() - 170 + "px");
+			ClickHandler closeHandler = closeEvent -> {
```
Comment (jay-hodgson) [inline src/main/java/org/sagebionetworks/web/client/widget/entity/tabs/FilesTabViewImpl.java:99]: replace with lambda expression

File: src/main/java/org/sagebionetworks/web/client/widget/entity/tabs/FilesTabViewImpl.java [hunk 2]
```diff
@@ -85,33 +87,35 @@ public FilesTabViewImpl(UserBadge createdByBadge, UserBadge modifiedByBadge) {
...
 
 	private ClickHandler getExpandClickHandler(final Widget w) {
-		return new ClickHandler() {
-			@Override
-			public void onClick(ClickEvent event) {
-				final Modal window = new Modal();
-				window.addStyleName("modal-fullscreen");
-				final ModalBody body = new ModalBody();
-				final Div oldParent = (Div) w.getParent();
+		return event -> {
```
Comment (jay-hodgson) [inline src/main/java/org/sagebionetworks/web/client/widget/entity/tabs/FilesTabViewImpl.java:90]: replace with lambda expression

File: src/main/java/org/sagebionetworks/web/client/widget/entity/tabs/FilesTabViewImpl.java [hunk 3]
```diff
@@ -65,6 +65,8 @@
 	Anchor expandProvenanceLink;
 	@UiField
 	Anchor expandPreviewLink;
+	@UiField
+	Div modalDialogContainer;
```
Comment (jay-hodgson) [inline src/main/java/org/sagebionetworks/web/client/widget/entity/tabs/FilesTabViewImpl.java:69]: Add the dynamically created Modal to this Div (on preview or provenance graph expand)

File: src/main/java/org/sagebionetworks/web/client/widget/entity/tabs/FilesTabViewImpl.java [hunk 4]
```diff
@@ -85,33 +87,35 @@ public FilesTabViewImpl(UserBadge createdByBadge, UserBadge modifiedByBadge) {
...
-				footer.add(closeButton);
-				window.add(footer);
-				window.show();
-			}
+				oldParent.add(w);
+				w.setHeight(DEFAULT_WIDGET_HEIGHT);
+				window.hide();
+			};
+			window.addAttachHandler(attachEvent -> {
+				if (!attachEvent.isAttached()) {
```
Comment (jay-hodgson) [inline src/main/java/org/sagebionetworks/web/client/widget/entity/tabs/FilesTabViewImpl.java:106]: when this is detached (loading another page!), then close the modal (clean up)

**Review by xschildw — APPROVED** [review]

**Pull request merged**